linera-faucet/server/src/tests.rs

```diff
@@ -88,114 +89,208 @@ async fn test_faucet_rate_limiting() {
-        .await
-        .is_ok());
-    assert!(root
-        .do_claim(AccountPublicKey::test_key(2).into())
-        .await
-        .is_err());
+    let result2 = root.do_claim(AccountPublicKey::test_key(1).into()).await;
+    assert!(result2.is_ok(), "First claim should succeed at time 1000");
+
+    // Test: immediate second claim should fail (rate limit)
+    let result3 = root.do_claim(AccountPublicKey::test_key(2).into()).await;
+    assert!(
+        result3.is_err(),
+        "Second immediate claim should fail due to rate limit"
+    );
+
+    // Test: at time 3000, more tokens should be available
     clock.set(Timestamp::from(3000));
-    assert!(root
-        .do_claim(AccountPublicKey::test_key(3).into())
-        .await
-        .is_ok());
-    assert!(root
-        .do_claim(AccountPublicKey::test_key(4).into())
-        .await
-        .is_ok());
-    assert!(root
-        .do_claim(AccountPublicKey::test_key(5).into())
-        .await
-        .is_err());
-    // If a validator is offline, it will create a pending block and then fail.
-    clock.set(Timestamp::from(6000));
-    builder.set_fault_type([0, 1], FaultType::Offline).await;
-    assert!(root
-        .do_claim(AccountPublicKey::test_key(6).into())
-        .await
-        .is_err());
-    assert_eq!(context.lock().await.update_calls, 4); // Also called in the last error case.
+    let result4 = root.do_claim(AccountPublicKey::test_key(3).into()).await;
+    assert!(result4.is_ok(), "Third claim should succeed at time 3000");
+
+    let result5 = root.do_claim(AccountPublicKey::test_key(4).into()).await;
+    assert!(result5.is_ok(), "Fourth claim should succeed at time 3000");
+
+    // Test: too many claims should eventually fail
+    let result6 = root.do_claim(AccountPublicKey::test_key(5).into()).await;
+    assert!(
+        result6.is_err(),
+        "Fifth claim should fail due to rate limit"
+    );
+
+    // Verify update_wallet calls (includes successful operations and final error case)
+    let update_calls = context.lock().await.update_calls;
+    assert_eq!(update_calls, 3);
+
+    // Clean up
+    cancellation_token.cancel();
+    let _ = processor_task.await;
+}
+
+#[tokio::test]
+async fn test_faucet_storage_basic() {
+    // Test basic storage file operations
+    use tempfile::tempdir;
+
+    let temp_dir = tempdir().unwrap();
+    let storage_path = temp_dir.path().join("test_storage.json");
+
+    // Test loading non-existent storage creates empty storage
+    let storage = super::FaucetStorage::load(&storage_path).await.unwrap();
+    assert!(storage.owner_to_chain.is_empty());
+
+    // Test saving empty storage
+    storage.save(&storage_path).await.unwrap();
+    assert!(storage_path.exists());
+
+    // Test loading saved empty storage
+    let loaded_storage = super::FaucetStorage::load(&storage_path).await.unwrap();
+    assert!(loaded_storage.owner_to_chain.is_empty());
+}
+
+#[test]
+fn test_multiply() {
```

Comment:
I wonder if there's a way to rewrite this to a prop test? maybe turn both numbers to u256 and multiply then compare with the result of `super::multiply`?

Reply:
Maybe, yes.

But this test wasn't even modified in this PR, only moved. I can move it back, but I do think it makes sense to have it in the `tests` module.